cmd/tailscaled/tailscaled.go

```diff
@@ -902,14 +886,30 @@ func applyIntegrationTestEnvKnob() {
 	}
 }
 
-func defaultEncryptState() bool {
+// defaultEncryptState reports whether we can enable state file encryption by
+// default, and returns an error if there's a concrete reason it can't be
+// enabled.
+func defaultEncryptState(statePath string) (bool, error) {
 	if runtime.GOOS != "windows" && runtime.GOOS != "linux" {
 		// TPM encryption is only configurable on Windows and Linux. Other
 		// platforms either use system APIs and are not configurable
 		// (Android/Apple), or don't support any form of encryption yet
 		// (plan9/FreeBSD/etc).
-		return false
+		return false, fmt.Errorf("--encrypt-state is not supported on %s", runtime.GOOS)
+	}
+	// Check if we have TPM support in this build.
+	if !store.HasKnownProviderPrefix(store.TPMPrefix + "/") {
+		return false, errors.New("--encrypt-state is not supported in this build of tailscaled")
 	}
+	// Check if we have TPM access.
+	if !hostinfo.New().TPM.Present() {
```

Comment:
feature.TPMAvailable() is much lighter

Reply:
Aren't they checking quite different things? AFAICT `hostinfo.New().TPM.Present()` checks whether the machine has a TPM available, whereas `feature.TPMAvailable()` just checks whether the binary was built with TPM support. Seems like we need to check both? We check the latter in a less direct way on the previous check, but I can change it to be based on the `feature` package instead.